samza-yarn/src/main/scala/org/apache/samza/job/yarn/YarnJob.scala

```diff
@@ -71,8 +71,18 @@ class YarnJob(config: Config, hadoopConfig: Configuration) extends StreamJob {
       )
     } catch {
       case e: Throwable =>
-        client.cleanupStagingDir
-        throw e
+        logger.error("Exception submitting yarn job.", e )
+        try {
+          // try to clean up. this may throw an exception depending on how far into launching the job we got.
+          // we don't want to mask the original problem by throwing this.
+          client.cleanupStagingDir
+        }
+        catch {
```

Comment:
Nitpick: Samza coding convention is to keep this on the previous line. Same for the finally. I.e.,
```
try {
...
} catch {
...
} finally {
...
}
```

Reply:
Sure thing. Done.